### PR TITLE
Sprint 2

### PR DESCRIPTION
**Pull request by EvelynJane**
Hi Ahsan, 

I managed to resolve all the issues from Sprint 1 and finish Sprint 2.

Thank you, and have a great weekend

### CODE REVIEW

File: src/test/java/Pages/Hotels/LaunchPage.java [hunk 1]
```diff
@@ -26,40 +26,101 @@
...
-    By getARewardNightIconLocator =
+    By freeCancellationTextLocator = By.xpath("//h3[text()='Free cancellation']");
+    By websiteFeedbackLocator = By.xpath("//div[@role='dialog']//a[text()='Website feedback']");
+
+
+    //TODO By priceGuaranteeIconLocator
+    By priceGuaranteeLinkLocator = By.xpath("//a[@href='/customer_care/pillar/price_guarantee.html?intlid=%5BInsight%5D+Trust+module+%3A%3A+PG+CTA+clicked']");
+    By priceGuaranteeIconLocator = By.cssSelector("main._24UiGk.home div._3umLRC div.IGZCbh section._2Npeyr:nth-child(3) ul._3TYstR._16C_8a:nth-child(2) > li.bPi4hj._2Y71AG._1geHg4");
+    //TODO By getARewardNightIconLocator
+    By getARewardNightLinkLocator = By.xpath("//a[@href='/hotel-rewards-pillar/hotelscomrewards.html?intlid=%5BInsight%5D+Trust+module+%3A%3A+HRW+CTA+clicked']");
```
Comment (ahsansiddiki) [inline src/test/java/Pages/Hotels/LaunchPage.java:46]: Above 3 locators doesn't look great, pls create custom xpath. You can use contains().

File: src/test/java/Pages/Hotels/Search.java [hunk 2]
```diff
@@ -21,14 +21,8 @@
     By checkOutDateLocator = By.xpath("//button[contains(@name,'Sun Apr 10 2022 12:00:00 GMT-0500 (Central Daylight Time')]");
```
Comment (ahsansiddiki) [inline src/test/java/Pages/Hotels/Search.java:21]: Hard coded date in locators isn't great. Pls try to create dynamic xpath.

File: src/test/java/Pages/Hotels/SignUpPage.java [hunk 3]
```diff
@@ -8,17 +8,18 @@
...
+    By tickMarkLocatorPassword = By.xpath("//div[@class='form-element valid']");
+    By tickMarkLocatorFirstName = By.xpath("//div[@class='form-element valid'][3]");
+    By tickMarkLocatorLastName = By.xpath("//div[@class='form-element valid'][4]");
+    String validEmail = "mznxbc@123.com";
     String validPassword = "asdfg12345";
-    String validFirstName= "Bingo";
+    String validFirstName = "Bingo";
     String validLastName = "Jones";
     String invalidEmail = "78@";
     String shortPassword = "123ab";
```
Comment (ahsansiddiki) [inline src/test/java/Pages/Hotels/SignUpPage.java:25]: Pls make SD dynamic and pass above test datas from feature files steps. 
Example:
@When(“^I enter (.+) in page comments$“)    public void enterComments(String data) {        fPage.pageComments(data);    }

File: src/test/resources/Hotels_/LaunchPage.feature [hunk 4]
```diff
@@ -1,9 +1,23 @@
...
-
-
+Background:Launch site
+  When I launch Website Hotels.com
 
   Scenario: Verify past date and back button on Current month's calendar is disabled
     When I Click on CheckIn Calendar
     And  I verify Past dates are disabled.
     Then I verify back button on current month is disabled
+    Then I close webpage
```
Comment (ahsansiddiki) [inline src/test/resources/Hotels_/LaunchPage.feature:9]: After first When/Then step following action/verification steps should be labelled as And

File: src/test/resources/Hotels_/Search.feature [hunk 5]
```diff
@@ -12,6 +13,7 @@ Feature: Search
     Then I click on “Search”button
```
Comment (ahsansiddiki) [inline src/test/resources/Hotels_/Search.feature:13]: Above step is an action step, should be labelled as And

**Comment by EvelynJane** [comment]
Thank you for the feedback, I did what I had to do in order to complete it,
but, yes I will work on better locators, making necessary items dynamic,
and my use of And When & Then
*Evelyn Jane Alexander*
631 374 5894

**Pull request merged**